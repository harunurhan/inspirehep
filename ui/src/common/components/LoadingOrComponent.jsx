import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Loading from './Loading';

class LoadingOrComponent extends Component {
  render() {
    if (this.props.loading) {
      return <Loading />;
    }
    return this.props.children;
  }
}

LoadingOrComponent.propTypes = {
  loading: PropTypes.bool,
  children: PropTypes.node,
};

LoadingOrComponent.defaultProps = {
  loading: undefined,
  children: undefined,
};

export default LoadingOrComponent;
