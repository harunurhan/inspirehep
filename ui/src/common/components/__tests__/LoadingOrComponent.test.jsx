import React from 'react';
import { shallow } from 'enzyme';

import LoadingOrComponent from '../LoadingOrComponent';

describe('LoadingOrComponent', () => {
  it('render with loading', () => {
    const wrapper = shallow(<LoadingOrComponent loading={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render without loading', () => {
    const wrapper = shallow(
      <LoadingOrComponent loading={false}>
        <div>
          <h2>Test</h2>
        </div>
      </LoadingOrComponent>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
