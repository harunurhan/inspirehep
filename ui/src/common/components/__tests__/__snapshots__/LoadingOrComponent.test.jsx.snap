// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOrComponent render with loading 1`] = `<Loading />`;

exports[`LoadingOrComponent render without loading 1`] = `
<div>
  <h2>
    Test
  </h2>
</div>
`;
